distributed/worker.py

```diff
@@ -1843,18 +1819,21 @@ def ensure_communicating(self):
                         worker = random.choice(local)
                     else:
                         worker = random.choice(list(workers))
-                    to_gather, total_nbytes = self.select_keys_for_gather(worker, dep)
+                    to_gather, total_nbytes = self.select_keys_for_gather(
+                        worker, dep.key
+                    )
                     self.comm_nbytes += total_nbytes
                     self.in_flight_workers[worker] = to_gather
                     for d in to_gather:
-                        self.transition_dep(d, "flight", worker=worker)
+                        self.transition(self.tasks.get(d), "flight", worker=worker)
```

Comment:
Is it safe to do `self.tasks[d]` here, or should we be concerned about `d not in self.tasks` as a possibility?

Reply:
I think it should be safe to do `self.tasks[d]` here, since the dependencies from `dep` should be created at this point.  I think I was worried about something dying in flight but that doesn't apply here.